### ISSUE TITLE
Melt tagger do not use its data_dir param

### ISSUE THREAD

**Issue by alexis-tonnoir** [sammous/spacy-lefff#27]
Hello,

thanks for the greate work, just got an issue trying to custumize dir where to download model :

melt_tagger.py init :

```
    def __init__(
            self,
            data_dir=DATA_DIR,
            lexicon_file_name=LEXICON_FILE,
            tag_file_name=TAG_DICT,
            print_probas=False):
        super(
            POSTagger,
            self).__init__(
            PACKAGE,
            url=URL_MODEL,
            download_dir=DATA_DIR)
```

data_dir=DATA_DIR is a parameter that is not used as it calls it super with download_dir=DATA_DIR.
Shouldn't it be download_dir=data_dir ?

I created a PR #28 

Regards.

**Comment by sammous** [comment]
oh indeed, thanks for spotting this typo!

**Comment by alexis-tonnoir** [comment]
Welcome :)
Do you know if/when it would be possible to deploy the fix ?

**Comment by alexis-tonnoir** [comment]
Hello again @sammous ,
Do you think you would have a chance to merge and deploy this little typo soon ? I have some constraints on machine on which i would like to use it, and it should help me a lot to be able to define this data path myself.
Many thanks.

**Comment by sammous** [comment]
hey sorry i will release it asap

**Comment by sammous** [comment]
i released a new version `0.3.8` with your update. Let me know if it solves your issue.
I should release a new proper version with spacy `>3.0.0`, as there is some breaking chnges with the new spacy version.

**Comment by alexis-tonnoir** [comment]
Hello @sammous ,
Many thanks, unfortunatly i missed a call to load_model function that do not use parameters but static definition that is not linked with the data_dir:

in __init__ you can pass the data_dir variable. then it dowloads the data. at line 119 it calls self.load_model() that will load model from MODELS_DIR = os.path.join(DATA_DIR, PACKAGE, 'models/fr') that doesn't exists if you downloaded data to a custom path.

I first made another fix PR : #30 

I check a little bit more into the code and get with few things i change in another PR #31 

1. the static variables related to data, lexicon,tag and model are used in the __init__ or dependent functions, and they all depend on the DATA_DIR. Maybe only the DATA_dir should be static and the other should be defined directly in the __init__. that avoid calling other function with lexicon or model dir that are no more consistent with data dir
-> i defined them in the __init__ function with override possibility this way :
```
def __init__(self, data_dir=DATA_DIR, model_dir_path=None,...)
model_dir_path = model_dir_path if model_dir_path else os.path.join(data_dir, package, 'models/fr')
```

2. as the __init__ function in the melt tagger call the load_model itself, is there any reason we would like to instanciate the POSTagger on a model and then load another model on same instance (and so pointing to data, lexicon and tag files that may not be consistent) ?
maybe this function has the meanning of being called into the __init__ only and so be private
-> I change it to _load_model(self, model_path)

3. the __init__ of the package is changing the logging level, so that when you got an application using logging and import spacy-leff it overrides your logger and set your log to debug.
-> i remove the logging override

4. the model dowload is done when you instanciate the POSTagger. it also takes few time to download and necessit you python running it to have some write access.
-> i created the __main__.py so that you can call it from command and download the data. you can download data this way :
python -m spacy_lefff download_tagger --download_dir /home/workspace/spacy_test

What do you think about this few changes ?

**Comment by sammous** [comment]
I released a new version `0.3.9` with the changes you did.
And additionally i adapted to spacy v3 and deprecated `python 2` for the new release `0.4.0`.
thanks for your contribution!

**Comment by alexis-tonnoir** [comment]
Hello,
I succesfuly succesfuly integrated it with spacy 3 on my project.
Thanks a lot.
